scripts/web/utils/pipeline.py

```
@@ -0,0 +1,101 @@
+
+def _shared_chroma_ready() -> bool:
+    chroma_dir = _shared_chroma_dir()
+    return chroma_dir.exists() and (chroma_dir / "chroma.sqlite3").exists()
+
+
+def _run(args: list[str], extra_env: dict | None = None) -> tuple[int, str]:
+    env = os.environ.copy()
+    if extra_env:
+        env.update(extra_env)
+    try:
+        proc = subprocess.run(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)